code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1488,3 +1488,28 @@ var/list/rank_prefix = list(\
 		src << SPAN_NOTICE("You can't do it right now.")
 	return
 
+/mob/living/carbon/human/should_have_organ(var/organ_check)
+
+	var/obj/item/organ/external/affecting
+	if(organ_check in list(O_HEART, O_LUNGS))
+		affecting = organs_by_name[BP_CHEST]
+	else if(organ_check in list(O_LIVER, O_KIDNEYS))
+		affecting = organs_by_name[BP_GROIN]
+
+	if(affecting && (affecting.robotic >= ORGAN_ROBOT))
+		return FALSE
+	return (species && species.has_organ[organ_check])
+
+/mob/living/carbon/human/has_limb(var/limb_check)	//returns TRUE if found, 2 if limb is robotic, FALSE if not found and null if its chest or groin (dont pass those)
+
+	if (limb_check == BP_CHEST || limb_check == BP_GROIN)	//obviously doesnt work with them
+		return
```

Comment:
> //obviously doesnt work with them
And what about `BP_HEAD`?
That check looks unobvious.

Reply:
then how should i name this check then ? has_appendage ?

Reply:
What that proc should do?
Why `BP_HEAD` counted as limb but `BP_GROIN` not?

Reply:
it should check if target has this body.
Arms, legs, head can be cut off

Reply:
@LagannPM groin can be cut off too